src/EnergyPlus/HVACFourPipeBeam.cc

```diff
@@ -0,0 +1,1302 @@
+// C++ Headers
+#include <cmath>
+
+// ObjexxFCL Headers
+#include <ObjexxFCL/Array.functions.hh>
+#include <ObjexxFCL/Fmath.hh>
+
+// EnergyPlus Headers
+#include <AirTerminalUnit.hh>
+#include <HVACFourPipeBeam.hh>
+#include <BranchNodeConnections.hh>
+#include <DataContaminantBalance.hh>
+#include <DataDefineEquip.hh>
+#include <DataEnvironment.hh>
+#include <DataHVACGlobals.hh>
+#include <DataIPShortCuts.hh>
+#include <DataLoopNode.hh>
+#include <DataPlant.hh>
+#include <DataPrecisionGlobals.hh>
+#include <DataSizing.hh>
+#include <DataZoneEnergyDemands.hh>
+#include <DataZoneEquipment.hh>
+#include <FluidProperties.hh>
+#include <General.hh>
+#include <GeneralRoutines.hh>
+#include <InputProcessor.hh>
+#include <NodeInputManager.hh>
+#include <OutputProcessor.hh>
+#include <PlantUtilities.hh>
+#include <Psychrometrics.hh>
+#include <ReportSizingManager.hh>
+#include <ScheduleManager.hh>
+#include <UtilityRoutines.hh>
+#include <CurveManager.hh>
+
+
+namespace EnergyPlus {
+
+namespace FourPipeBeam {
+
+	Array1D< std::shared_ptr< HVACFourPipeBeam > > FourPipeBeams; // dimension to number of machines
```

Comment:
I think the answer is yes, they are used in a structure defined in DataDefineEquip.hh

 

       struct ZoneAirEquip

 

              std::shared_ptr< AirTerminalUnit > airTerminalPtr;

 

This is how we can get to OO method calls from the rest of the program.  I took this pattern from Edwin’s branch PullLogicFromPlantComponents.  

 

But note the type there is the base case AirTerminalUnit, not these HVACFourPipeBeam objects, so that in the future other types of air terminal units can use the same thing. 

 

 

 

 

From: Mark Adams [mailto:notifications@github.com] 
Sent: Thursday, August 20, 2015 8:55 AM
To: NREL/EnergyPlus <EnergyPlus@noreply.github.com>
Cc: Brent Griffith <Brent.Griffith@EnergyArchmage.com>
Subject: Re: [EnergyPlus] Four pipe beam air terminal (#5129)

 

In src/EnergyPlus/HVACFourPipeBeam.cc <https://github.com/NREL/EnergyPlus/pull/5129#discussion_r37525101> :

> +#include <InputProcessor.hh>
> +#include <NodeInputManager.hh>
> +#include <OutputProcessor.hh>
> +#include <PlantUtilities.hh>
> +#include <Psychrometrics.hh>
> +#include <ReportSizingManager.hh>
> +#include <ScheduleManager.hh>
> +#include <UtilityRoutines.hh>
> +#include <CurveManager.hh>
> +
> +
> +namespace EnergyPlus {
> +
> +namespace FourPipeBeam {
> +
> +     Array1D< std::shared_ptr< HVACFourPipeBeam > > FourPipeBeams; // dimension to number of machines

@EnergyArchmage <https://github.com/EnergyArchmage>  Do you intend for these HVACFourPipeBeam objects to be used and referenced by other objects? Basically do these need to be std::shared_ptr or can they be std::unique_ptr? They very well could require the use of std::shared_ptr. 

For example, will an air terminal or any other object hold a std::shared_ptr< HVACFourPipeBeam > object within their class?

@Myoldmopar <https://github.com/Myoldmopar>  fyi

—
Reply to this email directly or view it on GitHub <https://github.com/NREL/EnergyPlus/pull/5129/files#r37525101> .  <https://github.com/notifications/beacon/AIWWUVouOp9RfAKrM7S_1Wc1_an-ObjWks5opcWigaJpZM4FvC16.gif>